offchain-modules/packages/scripts/src/integration-test/eth_batch_test.ts

```diff
@@ -0,0 +1,336 @@
+
+const ETH_NODE_URL = process.env.ETH_URL || 'http://127.0.0.1:8545';
+const ETH_PRI = process.env.ETH_PRIV_KEY || '0xc4ad657963930fbff2e9de3404b30a4e21432c89952ed430b56bf802945ed37a';
+
+const CKB_NODE_URL = process.env.CKB_URL || 'http://127.0.0.1:8114';
+const CKB_INDEXER_URL = process.env.CKB_INDEXER_URL || 'http://127.0.0.1:8116';
+const CKB_PRI = process.env.CKB_PRIV_KEY || '0xa800c82df5461756ae99b5c6677d019c98cc98c7786b80d7b2e77256e46ea1fe';
+
+// const FORCE_BRIDGE_URL = 'http://47.56.233.149:3083/force-bridge/api/v1';
+
+// const ETH_NODE_URL = 'https://rinkeby.infura.io/v3/48be8feb3f9c46c397ceae02a0dbc7ae';
+// const ETH_PRI = 'XXX';
+
+// const CKB_NODE_URL = 'https://testnet.ckbapp.dev';
+// const CKB_INDEXER_URL = 'https://testnet.ckbapp.dev/indexer';
+// const CKB_PRI = 'XXX';
+
+const ckb = new CKB(CKB_NODE_URL);
+
+const client = new JSONRPCClient((jsonRPCRequest) =>
+  fetch(FORCE_BRIDGE_URL, {
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json',
+    },
+    body: JSON.stringify(jsonRPCRequest),
+    id: 1,
+  }).then((response) => {
+    if (response.status === 200) {
+      // Use client.receive when you received a JSON-RPC response.
+      return response.json().then((jsonRPCResponse) => client.receive(jsonRPCResponse));
+    } else if (jsonRPCRequest.id !== undefined) {
+      return Promise.reject(new Error(response.statusText));
+    }
+  }),
+);
+
+async function generateLockTx(
+  ethWallet: ethers.Wallet,
+  assetIdent: string,
+  nonce: number,
+  recipient: string,
+  amount: string,
+): Promise<string> {
+  const lockPayload = {
+    sender: ethWallet.address,
+    recipient: recipient,
+    asset: {
+      network: 'Ethereum',
+      ident: assetIdent,
+      amount: amount,
+    },
+  };
+  const unsignedLockTx = await client.request('generateBridgeInNervosTransaction', lockPayload);
+  logger.info('unsignedMintTx', unsignedLockTx);
+
+  const provider = new ethers.providers.JsonRpcProvider(ETH_NODE_URL);
+
+  const unsignedTx = unsignedLockTx.rawTransaction;
+  unsignedTx.value = unsignedTx.value ? ethers.BigNumber.from(unsignedTx.value.hex) : ethers.BigNumber.from(0);
+  unsignedTx.nonce = nonce;
+  unsignedTx.gasLimit = ethers.BigNumber.from(1000000);
+  unsignedTx.gasPrice = await provider.getGasPrice();
+
+  logger.info('unsignedTx', unsignedTx);
+
+  const signedTx = await ethWallet.signTransaction(unsignedTx);
+  logger.info('signedTx', signedTx);
+
+  const hexTx = await Promise.resolve(signedTx).then((t) => ethers.utils.hexlify(t));
+  return hexTx;
+}
+
+async function generateBurnTx(
+  asset: string,
+  ckbPriv: string,
+  sender: string,
+  recipient: string,
+  amount: string,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+): Promise<any> {
+  const burnPayload = {
+    network: 'Ethereum',
+    sender: sender,
+    recipient: recipient,
+    asset: asset,
+    amount: amount,
+  };
+
+  for (let i = 0; i < 5; i++) {
```

Comment:
is the loop here for retry? There is no sleep between retry, I wonder whether it works if there is dead outpoint errors.

Reply:
this retry is in order to avoid that CKB testnet node rpc access restriction, most time it only need to retry twice, and there is no need to sleep, it's tested to be ok in testnet